core/src/main/java/com/amplitude/core/utilities/Diagnostics.kt

```diff
@@ -4,7 +4,11 @@ import java.util.Collections
 
 class Diagnostics() {
     private var malformedEvents: MutableList<String>? = null
-    private var errorLogs: MutableList<String>? = null
+    private var errorLogs: MutableSet<String>? = null
```

Comment:
DOes it make sense to initialize this always and clear the set every time we flush? Helps skip a bunch of null checks and initializations.

Reply:
Yeah, could be that way too, was hoping it could save a bit of memory though maybe not that much.